### PR TITLE
draft for pr etiquette

### PR DESCRIPTION
**Pull request by tarekziade**
Proposed PR etiquette

### CODE REVIEW

File: docs/pr_etiquette.md [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+tested, raise this with the submitter. Find out how they have tested it, and
+refuse the work if they have not. They may not have a mechanism to test it, in
+which case you may need to help.
+
+If, as a submitter, you know that this change is not fully tested, highlight
+this in the PR text, and talk to the reviewer.
+
+One exception is when the PR needs a review on a specific architecture.
+
+
```
Comment (timgrein) [inline docs/pr_etiquette.md:160]:
```suggestion
### Add the appropriate backport labels

Make sure to include the appropriate backport labels, if your PR needs to be backported to a past version.
```

Small suggestion to include backports explicitly

**Comment by artem-shelkovnikov** [comment]
It's nice, I like the text. I also think it's missing a more concrete/down-to-earth steps - maybe it's worth to put them to PR template though. Like a checklist of things for the submitter - these things have to be done on the PR to even be reviewed, these things improve your chances of having PR accepted, etc.

Same from the reviewer side - these things are not met? reject the PR. PR too big (more than +200, for example) or contains too many unrelated changes? reject the PR. Title or description are not clear? First ask for details, and then review once they are provided, etc.

File: docs/pr_etiquette.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+Ensure that your PR title is scannable. People will read through the list of
+PRs attached to a repo, and must be able to distinguish between them based on
+title. Include a story/issue reference if possible, so the reviewer can get any
+extra context. Include a reference to the subsystem affected, if this is a
+large codebase.
+
+
+### Use the Draft feature and proper labels
+
+If your PR is not fully ready yet for reviews, convert it to a `draft` so people
```
Comment (seanstory) [inline docs/pr_etiquette.md:58]: I'd add here to not add assignees yet if you're not ready to ask anyone to look at it.

File: docs/pr_etiquette.md [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+If you do this, you are no longer the reviewer of the PR. You are a
+collaborator, and so should not merge the PR.
+
+It is of course possible to find a new reviewer, but generally change will be
+speedier if you require the original submitter to fix the code themselves.
+Alternatively, if the original PR is 'good enough', raise the changes you'd
+like to see as separate stories/issues, and rework in your own PR.
+
+### It is not the reviewers responsibility to test the code
```
Comment (seanstory) [inline docs/pr_etiquette.md:142]: This probably holds up internally, but I'm not sure it does for prs submitted outside of our team. If a community member puts up a PR, I think we need to test it before merging.

**Review by seanstory — COMMENTED** [review]
I like the idea of having this content. However, I think it should be in the `contributing.md`, as per https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors.

`contributing.md` is a magically named file that Github uses to display in various parts of the UI to help contributors know what guidelines they should follow when contributing to your repo. This content fits that description.

File: docs/pr_etiquette.md (hunk 3 above)
Comment (tarekziade) [reply docs/pr_etiquette.md:142]: yeah we don't want to scare out contributors, you are right

**Comment by seanstory** [comment]
> By the way, why do we have all files UPPERCASE in `docs`? is there a specific intent or is this just to copy over `README` style files ?

They're listed all-caps [here](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/creating-a-default-community-health-file). But I think they work with any casing. IDK why the all-caps convention, but you're probably right that it stems from `README` convention. 🤷

**Review by seanstory — APPROVED** [review]

**Pull request merged**